pyresample/geometry.py

```diff
@@ -2000,8 +2000,14 @@ def get_area_slices(self, area_to_cover, shape_divisible_by=None):
 
             xstart = 0 if x[0] is np.ma.masked else x[0]
             ystart = 0 if y[1] is np.ma.masked else y[1]
-            xstop = self.width if x[1] is np.ma.masked else x[1] + 1
-            ystop = self.height if y[0] is np.ma.masked else y[0] + 1
+            if self.area_extent[0] > self.area_extent[2]:
+                xstop = self.width if x[0] is np.ma.masked else x[0] + 1
+            else:
+                xstop = self.width if x[1] is np.ma.masked else x[1] + 1
+            if self.area_extent[1] > self.area_extent[3]:
+                ystop = self.height if y[1] is np.ma.masked else y[1] + 1
+            else:
+                ystop = self.height if y[0] is np.ma.masked else y[0] + 1
```

Comment:
Don't you  need to swap `[x,y]start` also ?

Reply:
Indeed. Added.